### PR TITLE
IAM role notes for various KMS cloud APIs

### LINKED ISSUE #123: Document required IAM permissions for AWS KMS (but also for other providers)
I searched extensively and could not find any documentation about what AWS API calls (or GCE, or Azure, any other external provider really) are required for step-ca to work. Would be much easier to create an user/role with minimal permissions if a list of API calls was provided.

Also, there is no documentation about the naming, tagging, or anything related to the resources created, which also makes it hard to create properly scoped policies.

Thanks!

### PR DESCRIPTION
**Pull request by tashian**
Fixes https://github.com/smallstep/step-kms-plugin/issues/123

### CODE REVIEW

File: step-ca/configuration.mdx [hunk 1]
```diff
@@ -559,19 +559,22 @@ Please install the [`step kms` plugin](https://github.com/smallstep/step-kms-plu
 
 Also, make sure you have installed [the `gcloud` CLI](https://cloud.google.com/sdk/docs/install) and have [configured Google Cloud application default credentials](https://developers.google.com/accounts/docs/application-default-credentials) in your local environment, eg. by running `gcloud auth application-default login`.
 
-Next, let's generate a private key for your root CA inside Google Cloud KMS. Run:
+In the project of your choosing, first [create a Cloud KMS key ring](https://cloud.google.com/kms/docs/create-key-ring) called `step-ca`.
```
Comment (maraino) [inline step-ca/configuration.mdx]: `step-kms-plugin` can create the key ring if it does not exists.

File: step-ca/configuration.mdx [hunk 2]
```diff
@@ -682,6 +685,28 @@ When you start `step-ca`, it prints your X.509 root fingerprint, and the SSH use
...
+  * [`cloudkms.cryptoOperator`](https://cloud.google.com/iam/docs/understanding-roles#cloudkms.cryptoOperator)
+  * [`cloudkms.viewer`](https://cloud.google.com/iam/docs/understanding-roles#cloudkms.viewer)
+* More specifically, a running `step-ca` needs access to the following KMS APIs:
+  * `resourcemanager.projects.get`
+  * `cloudkms.locations.get`
+  * `cloudkms.keyRings.get`
+  * `cloudkms.cryptoKeys.get`
+  * `cloudkms.cryptoKeys.list`
+  * `cloudkms.cryptoKeyVersions.get`
+  * `cloudkms.cryptoKeyVersions.list`
```
Comment (maraino) [inline step-ca/configuration.mdx]: We never list keys.
```suggestion
  * `cloudkms.cryptoKeyVersions.get`
```

File: step-ca/configuration.mdx [hunk 3]
```diff
@@ -682,6 +685,28 @@ When you start `step-ca`, it prints your X.509 root fingerprint, and the SSH use
...
+See [Google Cloud KMS: Access Control with IAM](https://cloud.google.com/kms/docs/iam) for an overview.
+
+Here are some notes on granting a custom IAM role that provides step-ca with access to Google Cloud KMS:
+* You can constrain the role to the key ring that you created for step-ca.
+* For day-to-day operation, grant `step-ca` the following default roles:
+  * [`cloudkms.cryptoOperator`](https://cloud.google.com/iam/docs/understanding-roles#cloudkms.cryptoOperator)
+  * [`cloudkms.viewer`](https://cloud.google.com/iam/docs/understanding-roles#cloudkms.viewer)
+* More specifically, a running `step-ca` needs access to the following KMS APIs:
+  * `resourcemanager.projects.get`
+  * `cloudkms.locations.get`
```
Comment (maraino) [inline step-ca/configuration.mdx:698]: I guess this is ok, but I have not idea. We should test this.

File: step-ca/configuration.mdx [hunk 4]
```diff
@@ -682,6 +685,28 @@ When you start `step-ca`, it prints your X.509 root fingerprint, and the SSH use
 2022/09/20 16:28:45 Serving HTTPS on :443 ...
 ```
 
+#### Notes on Google Cloud KMS Permissions
```
Comment (maraino) [inline step-ca/configuration.mdx:686]: The permissions here are only for `step-ca`. They won't work for creating keys using `step-kms-plugin`. The same happens for Azure and AWS.

**Review by maraino — CHANGES_REQUESTED** [review]
We can get rid of the *list roles. This also doesn't fix the issue mentioned, as it is specifically for `step-kms-plugin` that can create keys.

**Review by maraino — COMMENTED** [review]
LGTM: It's clear in the docs that this is for `step-ca`. 

It would be nice to make sure this works and also add the roles for `step-kms-plugin`

**Comment by tashian** [comment]
@maraino I'll add roles for `step-kms-plugin` in a separate PR.
Could you approve this PR?
It's still saying changes requested.

**Review by maraino — APPROVED** [review]

**Pull request merged**